From f449258155b7bc6ec59941f2ae5fd3da6c6a3f77 Mon Sep 17 00:00:00 2001
From: Meng Wei <wei.meng@intel.com>
Date: Mon, 14 Jan 2019 10:29:09 +0800
Subject: [PATCH 1246/1418] media: intel-ipu4: [VIRT] Avoid double close of
 dmabuf FD.

This patch is to fix the ksys_close call for
camera streaming for SOS case where the dmabuf fd
is close by user space. Added check to only close
the dmabuf fd for virtualized case.

Change-Id: Id0c7fbdc8b81235c36f0340501a8c695ec3aaacf
Signed-off-by: Ong Hock Yu <ong.hock.yu@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ipu-psys.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-psys.c b/drivers/media/pci/intel/ipu-psys.c
index de69adb..345285c 100644
--- a/drivers/media/pci/intel/ipu-psys.c
+++ b/drivers/media/pci/intel/ipu-psys.c
@@ -543,9 +543,6 @@ static int ipu_psys_release(struct inode *inode, struct file *file)
 				kbuf->dbuf = NULL;
 				kbuf->db_attach = NULL;
 				dma_buf_put(dbuf);
-#if defined(CONFIG_VIDEO_INTEL_IPU_ACRN) && defined(CONFIG_VIDEO_INTEL_IPU_VIRTIO_BE)
-				ksys_close(kbuf->fd);
-#endif
 			} else {
 				if (kbuf->db_attach)
 					ipu_psys_put_userpages(
-- 
2.7.4

